cmake/Modules/ObsDefaults_macOS.cmake

```diff
@@ -0,0 +1,147 @@
+cmake_minimum_required(VERSION 3.20)
+
+# Build options
+set(CMAKE_OSX_ARCHITECTURES
+    "x86_64"
+    CACHE STRING "OBS build architecture for macOS - x86_64 required at least")
+set(CMAKE_OSX_DEPLOYMENT_TARGET
+    "10.13"
+    CACHE STRING "OBS deployment target for macOS - 10.13+ required")
+if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+  set(CMAKE_INSTALL_PREFIX
+      "${CMAKE_BINARY_DIR}/rundir"
+      CACHE STRING "Directory to install OBS after building" FORCE)
+endif()
+
+# Enable modern cmake policies
+if(POLICY CMP0009)
+  cmake_policy(SET CMP0009 NEW)
+endif()
+
+if(POLICY CMP0011)
+  cmake_policy(SET CMP0011 NEW)
+endif()
+
+if(POLICY CMP0025)
+  cmake_policy(SET CMP0025 NEW)
+endif()
```

Comment:
Like `cmake/Modules/ObsDefaults_Linux.cmake`, could we place the policies section closer to the top of the file?  Do the policies set for Linux and macOS not affect Windows?

Reply:
CMP0009 handles Symlinks, which is rather uncommon on Windows still; CMP0025 handles Apple Clang so macOS only.

CMP0011 sets the policy push behaviour to automatic (meaning included scripts can push policies onto the global stack). This was introduced way back in CMake 2.6.3, so might not even be necessary anymore.

Reply:
I think you can make symlinks on Windows via Git Bash, and probably via WSL.  You can also do it via mklink if you have admin privileges.  You're probably correct that it's not terribly common.

CMake 3.21 will warn if [CMP0011](https://cmake.org/cmake/help/v3.21/policy/CMP0011.html) is not set and use the OLD behavior, so if we want the new behavior, we have to set it.

I'll leave this to what you decide is best.